### PR TITLE
SNOW-920995 add CI configuration for regression tests

### PR DESCRIPTION
**Pull request by sfc-gh-akolodziejczyk**
### Description
Please explain the changes you made here.

### Checklist
- [ ] Code compiles correctly
- [ ] Run ``make fmt`` to fix inconsistent formats
- [ ] Run ``make lint`` to get lint errors and fix all of them
- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: bindings_test.go [hunk 1]
```diff
@@ -983,6 +984,7 @@ func TestFunctionParameters(t *testing.T) {
 	}
 
 	runDBTest(t, func(dbt *DBTest) {
+		dbt.exec("ALTER SESSION SET BIND_NULL_VALUE_USE_NULL_DATATYPE=false")
```
Comment (sfc-gh-pfus) [inline bindings_test.go:987]: It may be better to use `mustExec` and ensure that no error is returned.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      println("${scmInfo}")
+      env.GIT_BRANCH = scmInfo.GIT_BRANCH
+      env.GIT_COMMIT = scmInfo.GIT_COMMIT
+    }
+    params = [
+      string(name: 'svn_revision', value: 'main'),
+      string(name: 'branch', value: 'main'),
+      string(name: 'client_git_commit', value: scmInfo.GIT_COMMIT),
+      string(name: 'client_git_branch', value: scmInfo.GIT_BRANCH),
+      string(name: 'TARGET_DOCKER_TEST_IMAGE', value: 'go-centos7-go1.21'),
```
Comment (sfc-gh-pfus) [inline Jenkinsfile:17]: We still support older go version, shouldn't we use the oldest one?

File: Makefile [hunk 3]
```diff
@@ -16,7 +16,7 @@ test_teardown:
 	kill -9 $$(ps -ewf | grep hang_webserver | grep -v grep | awk '{print $$2}') || true
 
 test: deps test_setup
-	./ci/scripts/test_component.sh
+	./ci/scripts/execute_tests.sh
```
Comment (sfc-gh-pfus) [inline Makefile:19]: Do we need this change? If I remember correctly the name of this script is the same in all repos.

File: parameters.json.local [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (sfc-gh-pfus) [inline parameters.json.local:1]: Is it a file for jenkins? Maybe a name can be more significant?

File: bindings_test.go (hunk 1 above)
Comment (sfc-gh-akolodziejczyk) [reply bindings_test.go]: done

File: Jenkinsfile (hunk 2 above)
Comment (sfc-gh-akolodziejczyk) [reply Jenkinsfile]: changed to 1.19

**Review by sfc-gh-pfus — APPROVED** [review]

**Pull request merged**